src/benchmarks.zig

```diff
@@ -1,326 +1,388 @@
+                Duration => {
+                    // print column headers
+                    if (arg_i == 0) {
+                        try writer_average.print("{s}, min, max, mean, variance\n", .{def.name});
+                    }
+                    const mean = sum / iter_count;
+                    var variance: u64 = 0;
+                    for (runtimes.items(.result)) |runtime| {
+                        const d = if (runtime > mean) runtime - mean else mean - runtime;
+                        const d_sq = d *| d;
+                        variance +|= d_sq;
+                    }
+                    variance /= iter_count;
+                    // print column results
+                    try writer_average.print("{s}, {d}, {d}, {d}, {d}\n", .{ arg_name, min, max, mean, variance });
+                },
+                inline else => {
+                    // print column headers
+                    if (arg_i == 0) {
+                        try writer_average.print("{s}, ", .{def.name});
+                        inline for (U.fields, 0..) |field, i| {
+                            if (i == U.fields.len - 1) {
+                                // dont print trailing comma
+                                try writer_average.print("{s}_min, {s}_max, {s}_mean, {s}_variance", .{ field.name, field.name, field.name, field.name });
+                            } else {
+                                try writer_average.print("{s}_min, {s}_max, {s}_mean, {s}_variance, ", .{ field.name, field.name, field.name, field.name });
+                            }
+                        }
+                        try writer_average.print("\n", .{});
+                    }
+
+                    // print results
+                    try writer_average.print("{s}, ", .{arg_name});
+                    inline for (U.fields, 0..) |field, j| {
+                        const f_max = @field(max_s, field.name);
+                        const f_min = @field(min_s, field.name);
+                        const f_sum = @field(sum_s, field.name);
+                        const T = @TypeOf(f_sum);
+                        const n_iters = switch (@typeInfo(T)) {
+                            .Float => @as(T, @floatFromInt(iter_count)),
+                            else => iter_count,
+                        };
+                        const f_mean = f_sum / n_iters;
+
+                        var f_variance: T = 0;
+                        const x: std.MultiArrayList(runtime_type).Field = @enumFromInt(j);
+                        for (runtimes.items(x)) |f_runtime| {
+                            const d = if (f_runtime > f_mean) f_runtime - f_mean else f_mean - f_runtime;
+                            switch (@typeInfo(T)) {
+                                .Float => f_variance = d * d,
+                                else => f_variance +|= d *| d,
+                            }
+                        }
+                        f_variance /= n_iters;
+
```

Comment:
Is squared variance what we want? It's hard to interpret in the human table. For the CSV, we could leave it as  variance in the csv and square it later.

Reply:
Maybe I'm misunderstanding something, but there's no squared variance here? `d * d` *is* variance, where `d` is just deviation.
I discussed this with Brennan before, and I think that standard deviation is a better benchmark for us than variance. (sqrt `f_variance` after all of these computations)

Reply:
[74d9015](https://github.com/Syndica/sig/pull/316/commits/74d9015aec1985cf2d2d824107ae422aa98b7af5)